cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -1419,24 +1420,40 @@ runQueryDRepState
       , Cmd.consensusModeParams
       , Cmd.networkId
       , Cmd.drepKeys = drepKeys'
+      , Cmd.includeStake
       , Cmd.mOutFile
       } = conwayEraOnwardsConstraints eon $ do
   let localNodeConnInfo = LocalNodeConnectInfo consensusModeParams networkId nodeSocketPath
 
   let drepKeys = case drepKeys' of
                    All -> []
                    Only l -> l
-  drepCreds <- Set.fromList <$> mapM (firstExceptT QueryCmdDRepKeyError . getDRepCredentialFromVerKeyHashOrFile) drepKeys
+  drepCreds <- mapM (firstExceptT QueryCmdDRepKeyError . getDRepCredentialFromVerKeyHashOrFile) drepKeys
+
+  drepState <- runQuery localNodeConnInfo $ queryDRepState eon $ Set.fromList drepCreds
+
+  drepStakeDistribution <-
+    case includeStake of
+      Cmd.WithStake -> runQuery localNodeConnInfo $
+        queryDRepStakeDistribution eon (Set.fromList $ Ledger.DRepCredential <$> drepCreds)
+      Cmd.NoStake -> return mempty
 
-  drepState <- runQuery localNodeConnInfo $ queryDRepState eon drepCreds
   writeOutput mOutFile $
-    second drepStateToJson <$> Map.assocs drepState
+    drepStateToJson drepStakeDistribution <$> Map.assocs drepState
   where
-    drepStateToJson ds = A.object
-      [ "expiry" .= (ds ^. Ledger.drepExpiryL)
-      , "anchor" .= (ds ^. Ledger.drepAnchorL)
-      , "deposit" .= (ds ^. Ledger.drepDepositL)
-      ]
+    drepStateToJson stakeDistr (cred, ds) = (cred,) . A.object $
```

Comment:
This is fine for now but we should convert this into a `ToJSON` instance.

Reply:
Follow-up issue created: https://github.com/IntersectMBO/cardano-cli/issues/606